pkg/openstackconfiggenerator/configmap.go

```diff
@@ -347,6 +347,23 @@ func createNetworksMap(
 	return networksMap, networkMappingList, nil
 }
 
+// IsRoleIncluded - checks if the role exists in the ConfigGenerator Roles set
+func IsRoleIncluded(roleName string, instance *ospdirectorv1beta1.OpenStackConfigGenerator) bool {
+
+	if len(instance.Spec.Roles) == 0 {
```

Comment:
could combine in one if for check on len and ctlplane role?

Reply:
done